src/python/pants/engine/scheduler.py

```diff
@@ -406,15 +406,25 @@ def visualize_graph_to_file(self, filename):
   def visualize_rule_graph_to_file(self, filename):
     self._scheduler.visualize_rule_graph_to_file(filename)
 
+  def execution_request_literal(self, request_specs):
```

Comment:
I think that rather than adding a new method, we'd want to just update `execute_request`... it only has two callers outside of unit tests. But that would increase the scope here... if you're able to swap out the methods as a quick followup (not here, as this is already a bit large), that would be appreciated.

Also, the `SchedulerSession.product_request` and `product_requests` family is easier to use... so adding a another method that is shaped like "list of tuples" would be ideal.

Reply:
That all sounds like a viable scope for a followup -- I'll make an issue.

Reply:
Made into #6044!